### DISCUSSION TITLE
Error getting pod controlled by the job while scanning vulnerabilityreports

### DISCUSSION THREAD

**Discussion by JosebaAndres** [aquasecurity/starboard#1396] [Questions & Help]
Hello,

I’ve created a new cluster using k3d to test the tool, and I followed the getting started guide step-by-step. However, I encounter this error when scanning the nginx deployment:

- Command: starboard scan vulnerabilityreports deployment/nginx
- Error: error: getting pod controlled by job: "starboard/scan-vulnerabilityreport-75d4f7c66": pod for job not found

I’ve saved the pod logs in a txt file (attached) and it seems that the scan completed successfully...

Does anyone know what might be causing this error?
[logs.txt](https://github.com/user-attachments/files/16727636/logs.txt)

Thanks and best regards,

**Comment by rsys-psaya** [comment]
Even i am observing the same issue in my k8s cluster which is deployed using k0s.

error: getting pod controlled by job: "starboard/scan-vulnerabilityreport-75d4f7c66": pod for job not found

Full log:

[root@psaya-wa7xix bin]# starboard scan vulnerabilityreports deployment/nginx -v=4
I0125 12:52:49.124223 1641934 scanner.go:67] Getting Pod template for workload: {Deployment nginx default}
I0125 12:52:49.131672 1641934 scanner.go:94] Scanning with options: {ScanJobTimeout:0s DeleteScanJob:true}
I0125 12:52:49.142380 1641934 runner.go:79] Running task and waiting forever
I0125 12:52:49.142447 1641934 runnable_job.go:74] Creating job "starboard/scan-vulnerabilityreport-75d4f7c66"
I0125 12:52:49.170829 1641934 reflector.go:219] Starting reflector *v1.Job (30m0s) from pkg/mod/k8s.io/client-go@v0.24.4/tools/cache/reflector.go:167
I0125 12:52:49.170858 1641934 reflector.go:255] Listing and watching *v1.Job from pkg/mod/k8s.io/client-go@v0.24.4/tools/cache/reflector.go:167
I0125 12:52:49.170856 1641934 reflector.go:219] Starting reflector *v1.Event (30m0s) from pkg/mod/k8s.io/client-go@v0.24.4/tools/cache/reflector.go:167
I0125 12:52:49.170882 1641934 reflector.go:255] Listing and watching *v1.Event from pkg/mod/k8s.io/client-go@v0.24.4/tools/cache/reflector.go:167
I0125 12:52:49.210607 1641934 runnable_job.go:130] Event: Created pod: scan-vulnerabilityreport-75d4f7c66-7m7gm (SuccessfulCreate)
I0125 12:52:49.271363 1641934 shared_informer.go:285] caches populated
I0125 12:52:49.271415 1641934 shared_informer.go:285] caches populated
I0125 12:53:11.824964 1641934 runnable_job.go:109] Stopping runnable job on task completion with status: Complete
I0125 12:53:11.825034 1641934 runner.go:83] Stopping runner on task completion with error: <nil>
I0125 12:53:11.825062 1641934 scanner.go:133] Scan job completed: starboard/scan-vulnerabilityreport-75d4f7c66
I0125 12:53:11.825102 1641934 scanner.go:157] Getting logs for nginx container in job: starboard/scan-vulnerabilityreport-75d4f7c66
I0125 12:53:11.831312 1641934 runnable_job.go:130] Event: Job completed (Completed)
I0125 12:53:11.839416 1641934 scanner.go:126] Deleting scan job: starboard/scan-vulnerabilityreport-75d4f7c66
error: getting pod controlled by job: "starboard/scan-vulnerabilityreport-75d4f7c66": pod for job not found


[root@psaya-wa7xix bin]# starboard get vulnerabilityreports deployment/nginx -o yaml
No reports found in default namespace.
